staging/src/k8s.io/kube-scheduler/config/v1alpha2/types.go

```diff
@@ -202,3 +218,20 @@ type PluginConfig struct {
 	// Args defines the arguments passed to the plugins at the time of initialization. Args can have arbitrary structure.
 	Args runtime.RawExtension `json:"args,omitempty"`
 }
+
+func (c *PluginConfig) decodeNestedObjects(d runtime.Decoder) error {
+	gvk := SchemeGroupVersion.WithKind(c.Name + "Args")
+	obj, parsedGvk, err := d.Decode(c.Args.Raw, &gvk, nil)
```

Comment:
Can `c.Args.Raw` be empty slice? If so, `err` would be a plain one with message "empty data".